### PR TITLE
[V3] Remove contacts and rename to preferences

### PR DESCRIPTION
**Pull request by nplasterer**
Clean up the contacts part of the sdk to be clearer about it being preferences.

### CODE REVIEW

File: library/src/main/java/org/xmtp/android/library/Conversations.kt [hunk 1]
```diff
@@ -316,11 +314,11 @@ data class Conversations(
...
-        val contacts = client.contacts
-        contacts.refreshConsentList()
-        if (contacts.consentList.state(peerAddress) == ConsentState.UNKNOWN) {
-            contacts.allow(listOf(peerAddress))
-        }
+//        val contacts = client.preferences
+//        contacts.refreshConsentList()
+//        if (contacts.consentList.state(peerAddress) == ConsentState.UNKNOWN) {
+//            contacts.allow(listOf(peerAddress))
+//        }
```
Comment (codabrink) [inline library/src/main/java/org/xmtp/android/library/Conversations.kt:321]: Do we want to remove these lines?

File: library/src/main/java/org/xmtp/android/library/PrivatePreferences.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    ALLOWED,
+    DENIED,
+    UNKNOWN;
+
+    companion object {
+        fun toFfiConsentState(option: ConsentState): FfiConsentState {
+            return when (option) {
+                ALLOWED -> FfiConsentState.ALLOWED
+                DENIED -> FfiConsentState.DENIED
+                else -> FfiConsentState.UNKNOWN
```
Comment (codabrink) [inline library/src/main/java/org/xmtp/android/library/PrivatePreferences.kt:17]: I ran into this in Rust too when matching between the proto and native consent state and was tempted to do `_ -> UNKNOWN`, but it's probably better to be explicit and do `UNKNOWN -> UNKNOWN` so this isn't a footgun in the future when we might add another consent state and the compiler happily matches it to `UNKNOWN`.

File: library/src/main/java/org/xmtp/android/library/PrivatePreferences.kt [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            }
+        }
+
+        fun fromFfiConsentState(option: FfiConsentState): ConsentState {
+            return when (option) {
+                FfiConsentState.ALLOWED -> ALLOWED
+                FfiConsentState.DENIED -> DENIED
+                else -> UNKNOWN
```
Comment (codabrink) [inline library/src/main/java/org/xmtp/android/library/PrivatePreferences.kt:25]: Same here.

**Review by codabrink — APPROVED** [review]
Just the comments, otherwise great!

File: library/src/main/java/org/xmtp/android/library/Conversations.kt (hunk 1 above)
Comment (nplasterer) [reply library/src/main/java/org/xmtp/android/library/Conversations.kt:321]: Good call this should be handled in libxmtp now.

**Pull request merged**